Dateï¼š20250910 & 20250917  
Languageï¼š[EN](#Fantastic-Genomic-Biomarkers-and-Where-to-Find-Them-Practical-Course-part-II) / [ä¸­æ–‡](#ç”Ÿç‰©æ¨™è¨˜ç‰©èˆ‡å®ƒå€‘çš„ç”¢åœ°å¯¦ä½œèª²ç¨‹äºŒ)  


# Fantastic Genomic Biomarkers and Where to Find Them Practical Course (Part II)

## Main Content of This Course
1. Copy course files to your folder on the National Center for High-Performance Computing (NCHC) system
2. **Sample QC**: A rapid quality control check on raw sequencing data. Through Quality Control, researchers can identify potential issues in the data, allowing them to make decisions about data filtering or correction. In high-throughput sequencing experiments, quality control is the process of inspecting and evaluating raw sequencing data to ensure data integrity and reliability.


## Copying Course Files
> [!CAUTION]
> Since files are being copied to NCHC, the following steps must be executed in your own remote host's "/work/{your_username}" path (**Please ensure you are in your own path!!!**)

## The File Tree of This Course
[FastQC_file tree](https://github.com/Jacob-s-Lab/2025-Biomarkers/blob/main/week2_0910_0917/wk2_tree.txt)

### Step 1: Create a Path on NCHC
1. Log in to NCHC (for those who forgot how to log in, please refer to this [link](https://hackmd.io/jcvG9iIiRW6DTUysi8AKug)).
2. Enter the work directory and type `cd /work/username`, then type `mkdir result` to create a folder named result in the current location on the NCHC system. This will be the folder where you store your course files.
    ```
    cd /work/username
    mkdir result
    ```
3. Next, use cd result and create a folder with mkdir fastq to store the downloaded samples.
    ```
    cd result
    mkdir fastqc
    ```

> [!IMPORTANT]
> #### Command Lecture
> `mkdir`
> - Make directory, used to create a new directory (folder) at the specified location.
> - Usage of mkdir: mkdir [options] # Create a new directory (folder)

4. Use the rsync command to copy the required scripts (bash file) and samples to your path:\
    **â—In this step, you need to learn how to copy files from someone else's folder on the NCHC to your own folder on the NCHC.â—**
    ```
    rsync -avz /work/evelyn92/2025Biomarker/fastqc.sh ./
    ```
5. Additionally, [download](https://paylong2.tw/nextcloud/index.php/s/NP76ri2MNpoasmq) the data required for analysis.
    ```
    password: NGS112-2
    ```
> â—**need to download six files!!** (6 reads in total, which include three samples) \
> SRR13076392_S14_L002_R1_001.fastq.gz \
> SRR13076392_S14_L002_R2_001.fastq.gz \
> SRR13076393_S15_L002_R1_001.fastq.gz \
> SRR13076393_S15_L002_R2_001.fastq.gz \
> SRR13076396_S18_L002_R1_001.fastq.gz I am running a few minutes late; my previous meeting is running over.
> SRR13076396_S18_L002_R2_001.fastq.gz

6. [Upload](https://github.com/Jacob-s-Lab/2025-Biomarkers/blob/main/tutorial%20week1.md) them to your own **fastqc folder**.\
    **â—In this step, you need to learn how to upload files from local to NCHC.â—**
    ```
    Upload files:rsync -azrvh . supercomputeracount@t3-c4.nchc.org.tw:/work/supercompputeraccount/result/fastqc
    unzip a file: unzip <filename_to_unzip>
    rename a file: mv <original_filename> <new_filename>
    ```
> [!CAUTION]
> - Folder and File Description (strongly recommended to understand):
>   `fastqc`: This is the directory you are currently in, containing all <ins>analysis execution files, reference sequences, and various tools</ins>. 

> [!IMPORTANT]
> #### File Format Introduction
> * `fastq` is a file format used to store DNA or RNA sequence data generated by high-throughput sequencing technologies in the biological field. This format includes both nucleotide sequences and corresponding quality scores.
> * `fastqc.sh`: This file is the executable required for the analysis.

---------------

## Sample QC
Shell script for running [FastQC](https://github.com/Jacob-s-Lab/2025-Biomarkers/blob/main/week2_0910_0917/fastqc.sh).

### Step 1: Create a Shell Script

> [!IMPORTANT]
> #### What is a Shell Script?
> ***Shell script*** is a program written using the functionality of the shell. It is a plain text file containing shell syntax and commands (including external commands), combined with regular expressions, pipeline commands, and data redirection, to achieve the desired processing. Writing and using shell scripts can improve work efficiency and reduce repetitive command operations. They can run in Unix, Linux, or other Unix-like systems' shell environments. Common shell scripting languages include Bash (Bourne Again Shell), Sh (Bourne Shell), and Zsh (Z Shell).
>
> Reference: https://linux.vbird.org/linux_basic/centos7/0340bashshell-scripts.php#script

1. Enter the result directory with 
   ```
    cd /work/username/result
    ```

2. Create a shell script by typing 
    ```
    vim fastqc.sh
    ```

> [!IMPORTANT]
> #### `vim fastqc.sh`
> - In Unix and Linux systems, the command `vim fastqc.sh` is used to open or create a file named `fastqc.sh` using the Vim editor. This is a simple command to enter the Vim editor to edit the specified Shell Script file.
> - Specific Meaning:
>     - vim: This is the name of the command-line text editor Vim. Vim is a powerful text editor widely used for editing program code and scripts.
>     - fastqc.sh: This is the file name to be edited. In this case, `fastqc.sh` is a Shell Script file, with the .sh extension typically indicating it is a Shell script file.
> - Behavior after executing this command:
>     - If the `fastqc.sh` file already exists, `vim fastqc.sh` will open the file, allowing you to view and edit its contents.
>     - If the `fastqc.sh` file does not exist, `vim fastqc.sh` will create a new empty file and enter Vim editor so you can start writing the script.
  
  
3. Press the <kbd>i</kbd> key to enter insert mode (you will see "â€“- INSERT â€“-" at the bottom) and paste the shell script you downloaded into `fastqc.sh`.

### Step 2: Modify the Analysis Script

1. Change the following code:
> The example below uses the SEA folder in the fastq directory (format should follow the provided example, excluding the file extension).

(1) SLURM Scheduling Settings

> [!IMPORTANT]
> #### What is SLURM?
> SLURM (Simple Linux Utility for Resource Management) is an open-source resource manager and workload scheduler for large-scale computing clusters. It is primarily used in high-performance computing (HPC) environments to manage and schedule computing resources such as CPUs, memory, and compute nodes. SLURM is widely used in large supercomputing centers, research institutions, and enterprises.


  ğŸ‘‰ Modify this block according to the instructions (see the explanation below): 

  ```
  #!/usr/bin/sh
  #SBATCH -A ACD114093              # Account name/project number
  #SBATCH -J fastqc                 # Job name
  #SBATCH -p ngscourse              # Partition Name (equivalent to PBS's -q Queue name)
  #SBATCH -c 2                      # Number of cores used (refer to Queue resource settings)
  #SBATCH --mem=13g                 # Amount of memory used (refer to Queue resource settings)
  #SBATCH -o out.log                # Path to the standard output file
  #SBATCH -e err.log                # Path to the standard error output file
  #SBATCH --mail-user=yourmail@gmail.com    # Email
  #SBATCH --mail-type=END           # Specifies when to send email; can be NONE, BEGIN, END, FAIL, REQUEUE, ALL
  # For NCHC usage
  ```

(2) Modify file path
  ```
  # Please enter the R1 & R2 file name and your username
  sampleR1=/work/username/result/fastqc/SRR13076392_S14_L002_R1_001.fastq.gz    # file path of the compressed reads
  sampleR2=/work/username/result/fastqc/SRR13076392_S14_L002_R2_001.fastq.gz    # file path of the compressed reads
  ```
(3) create a directory named fastqc_S14 to store the FastQC results
  ```
  mkdir fastqc_S14
  ```

(4) Modify the command for running FastQC
  ```
  fastqc ${sampleR1} ${sampleR2} -o fastqc_S14
  ```
(5) Press <kbd>esc</kbd> to exit insert mode.    
(6) Type `:wq` and press <kbd>Enter</kbd> to save and exit    
**â—if you see "E45: 'readonly' option is set (add ! to override)", type `:wq!` to save.â—**

> [!IMPORTANT]
> #### Commands Lecture
> - `:wq`: Save and exit. This command in Vim editor saves the current file and exits insert mode. Press Esc to enter command mode, then type`:wq` and press Enter. This saves the current file's modifications and exits Vim. "w" stands for write (save), "q" stands for quit (exit).
> - `:wq!`: Force save and exit. If the file is read-only or has other restrictions, you can use `:wq!` to force save and exit. "!" signifies force execution.
> - If you only want to save but not exit, use `:w`. To exit without saving, use `:q` or `:q!` (force exit).

2. Execute the Script

(1) Submit the edited script as a SLURM job with the following command:
```
sbatch fastqc.sh
```

> [!IMPORTANT]
> #### Commands Lecture 
> `sbatch` is a command-line tool for submitting job scripts to the SLURM job scheduling system. These scripts typically include SLURM directives and commands to be executed.

(2) If submission is successful, you will see:
```
Submitted batch job ï¼¿ï¼¿ï¼¿
```

(3) Use the following command to check the job status:
```
sacct
```
![image](https://hackmd.io/_uploads/Bkevfu630.png)

> [!IMPORTANT]
> #### Commands Lecture
> `sacct`
This command is used to list the status of jobs or job arrays associated with an account, such as running, terminated, or completed. It is the most basic command for viewing job statuses. It can display information such as ID, user, status, and resources used, which is useful for tracking and analyzing the status of jobs.

3. Entering `ls` allows you to view all the files in a directory, where you can see `out.log`, `err.log`, and HTML files.
- out.log and err.log are the standard output and standard error for this script. If any errors occurred during execution, you can check err.log.
- This script will generate an HTML file, which can be downloaded and opened to view the FastQC Report.
Details available for [download](https://github.com/Jacob-s-Lab/2025-Biomarkers/blob/main/tutorial%20week1.md) at the link

---------------
---------------

# ç”Ÿç‰©æ¨™è¨˜ç‰©èˆ‡å®ƒå€‘çš„ç”¢åœ°å¯¦ä½œèª²ç¨‹(äºŒ)


## æœ¬æ¬¡èª²ç¨‹ä¸»è¦å…§å®¹
 1. è¤‡è£½èª²ç¨‹æª”æ¡ˆè‡³åœ‹ç¶²ä¸­è‡ªå·±çš„è³‡æ–™å¤¾
 2. sample QC:å°åŸå§‹æ¸¬åºæ•¸æ“šé€²è¡Œå¿«é€Ÿè³ªé‡æ§åˆ¶æª¢æŸ¥ã€‚é€šéQuality Controlç ”ç©¶äººå“¡å¯ä»¥è­˜åˆ¥æ•¸æ“šä¸­çš„æ½›åœ¨å•é¡Œï¼Œå¾è€Œåšå‡ºæ•¸æ“šéæ¿¾ã€ä¿®æ­£çš„æ±ºå®šã€‚åœ¨é«˜é€šé‡æ¸¬åºå¯¦é©—ä¸­ï¼Œè³ªé‡æ§åˆ¶æ˜¯å°åŸå§‹æ¸¬åºæ•¸æ“šé€²è¡Œæª¢æŸ¥å’Œè©•ä¼°çš„éç¨‹ï¼Œä»¥ç¢ºä¿æ•¸æ“šçš„å®Œæ•´æ€§å’Œå¯é æ€§ã€‚


## è¤‡è£½èª²ç¨‹æª”æ¡ˆ
> [!CAUTION]
> â€¼ï¸å› ç‚ºæ˜¯å°‡æª”æ¡ˆè¤‡è£½è‡³åœ‹ç¶²ï¼Œæ‰€ä»¥ä»¥ä¸‹æ­¥é©Ÿçš„æŒ‡ä»¤éƒ½éœ€åœ¨ã€ä½ è‡ªå·±çš„ã€é ç«¯ä¸»æ©Ÿçš„ `/work/{your_username}` è·¯å¾‘ä¸‹ (**è«‹ç¢ºä¿ç¾åœ¨æ˜¯åœ¨è‡ªå·±çš„è·¯å¾‘ä¸‹ï¼ï¼ï¼**)

## æœ¬æ¬¡èª²ç¨‹çš„æ¨¹ç‹€è³‡æ–™çµæ§‹
[FastQC_file tree](https://github.com/Jacob-s-Lab/2025-Biomarkers/blob/main/week2_0910_0917/wk2_tree.txt)

### step 1 åœ¨åœ‹ç¶²ä¸Šå»ºç«‹è·¯å¾‘
1. ç™»å…¥åœ‹ç¶²ï¼ˆå¿˜è¨˜æ€éº¼ç™»å…¥çš„äººè«‹åƒè¦‹[é€£çµ](https://hackmd.io/jcvG9iIiRW6DTUysi8AKug)ï¼‰
2. é€²å…¥workè³‡æ–™å¤¾è¼¸å…¥`cd /work/username`ï¼Œæ¥è‘—è¼¸å…¥`mkdir result`å¯ä»¥åœ¨åœ‹ç¶²ä¸»æ©Ÿç›®å‰çš„ä½ç½®ä¸‹å»ºç«‹ä¸€å€‹å«åšresultçš„è³‡æ–™å¤¾ï¼Œä½œç‚ºæœ¬æ¬¡ä½œæ¥­æª”æ¡ˆå„²å­˜çš„è³‡æ–™å¤¾
    ```
    cd /work/username
    mkdir result
    ```
3. æ¥è‘—`cd result`åœ¨å»ºç«‹ä¸€å€‹è³‡æ–™å¤¾`mkdir fastqc`å„²å­˜ä¸‹è¼‰çš„sample
    ```
    cd result
    mkdir fastqc
    ```

> [!IMPORTANT]
> #### å‘½ä»¤å°å­¸å ‚
> - `mkdir` make directoryï¼Œç”¨ä¾†åœ¨æŒ‡å®šä½ç½®å‰µå»ºä¸€å€‹æ–°çš„ç›®éŒ„ï¼ˆè³‡æ–™å¤¾ï¼‰
> - mkdir çš„ç”¨æ³•: ```mkdir [é¸é …]   #å»ºç«‹ä¸€å€‹æ–°ç›®éŒ„(è³‡æ–™å¤¾)```

4. åœ¨terminalåˆ©ç”¨`rsync`æŒ‡ä»¤ï¼Œå°‡åˆ†ææ‰€éœ€çš„æŒ‡ä»¤(bashæª”)ï¼Œè¤‡è£½åˆ°è‡ªå·±çš„è·¯å¾‘ä¸‹ä½¿ç”¨\
    **â—æ­¤æ­¥é©Ÿä½ éœ€è¦å­¸æœƒå¦‚ä½•å¾åœ‹ç¶²åˆ¥äººçš„è³‡æ–™å¤¾ä¸­è¤‡è£½æª”æ¡ˆåˆ°è‡ªå·±åœ¨åœ‹ç¶²çš„è³‡æ–™å¤¾â—**
    ```
    rsync -avz /work/evelyn92/2025Biomarker/fastqc.sh ./
    ```

5. å¦å¤–[ä¸‹è¼‰](https://paylong2.tw/nextcloud/index.php/s/NP76ri2MNpoasmq)åˆ†ææ‰€éœ€è³‡æ–™
    ```
    #ä¸‹è¼‰å¯†ç¢¼:NGS112-2
    ```
   >â— **è«‹ä¸‹è¼‰ä»¥ä¸‹å…­å€‹æª”æ¡ˆ!!** (6æ¢readsï¼Œå…±åŒ…å«ä¸‰å€‹sample) \
   > SRR13076392_S14_L002_R1_001.fastq.gz \
   > SRR13076392_S14_L002_R2_001.fastq.gz \
   > SRR13076393_S15_L002_R1_001.fastq.gz \
   > SRR13076393_S15_L002_R2_001.fastq.gz \
   > SRR13076396_S18_L002_R1_001.fastq.gz \
   > SRR13076396_S18_L002_R2_001.fastq.gz
   
6. è«‹ç›´æ¥[ä¸Šå‚³](https://github.com/Jacob-s-Lab/2025-Biomarkers/blob/main/tutorial%20week1.md)  æª”æ¡ˆåˆ°fastqc\
    **â—æ­¤æ­¥é©Ÿéœ€è¦å­¸ç¿’çš„å…§å®¹ç‚ºå°‡æª”æ¡ˆå¾æœ¬åœ°ç«¯ä¸Šå‚³è‡³åœ‹ç¶²â—**
    ```
    ä¸Šå‚³æª”æ¡ˆ:rsync -azrvh . ä¸»æ©Ÿå¸³è™Ÿ@t3-c4.nchc.org.tw:/work/ä¸»æ©Ÿå¸³è™Ÿ/result/fastqc
    è§£å£“ç¸®æª”æ¡ˆ: unzip <è¦è§£å£“ç¸®çš„æª”å>
    æ”¹æª”å: mv <åŸæœ¬çš„æª”å> <å¾Œä¾†çš„æª”å>
    ```

> [!CAUTION]
> `fastqc`ï¼šæ­¤**è³‡æ–™å¤¾**å­˜æ”¾<ins> ä¸‹è¼‰çš„æª”æ¡ˆ (fastqæª”æ¡ˆ) </ins>

> [!IMPORTANT]
> #### æ–‡ä»¶æ ¼å¼ä»‹ç´¹
> - `fastq` : FASTQ æ˜¯ä¸€ç¨®æ–‡ä»¶æ ¼å¼ï¼Œç”¨æ–¼å­˜å„²ç”Ÿç‰©å­¸é ˜åŸŸä¸­é«˜é€šé‡æ¸¬åºæŠ€è¡“ç”Ÿæˆçš„ DNA æˆ– RNA åºåˆ—æ•¸æ“šã€‚é€™ç¨®æ ¼å¼åŒæ™‚åŒ…å«äº†åºåˆ—çš„æ ¸è‹·é…¸åºåˆ—å’Œå°æ‡‰çš„è³ªé‡åˆ†æ•¸
> - `fastqc.sh` : æ­¤**æª”æ¡ˆ**ç‚ºåŸ·è¡Œåˆ†ææ‰€éœ€çš„åŸ·è¡Œæª”
     
---------------

# Sample QC
Shell script for running [FastQC](https://github.com/Jacob-s-Lab/2025-Biomarkers/blob/main/week2_0910_0917/fastqc.sh).

## Step 1 å»ºç«‹shell script

> [!IMPORTANT]
> #### ç”šéº¼æ˜¯shell script?
> shell  script(ç¨‹å¼åŒ–è‰ç¨¿)
> - ç°¡å–®ä¾†èªªï¼Œæ˜¯åˆ©ç”¨ shell çš„åŠŸèƒ½æ‰€å¯«çš„ä¸€å€‹ã€ç¨‹å¼ (program)ã€ï¼Œé€™å€‹ç¨‹å¼æ˜¯ä½¿ç”¨ç´”æ–‡å­—æª”ï¼Œå°‡ä¸€äº› shell çš„èªæ³•èˆ‡æŒ‡ä»¤(å«å¤–éƒ¨æŒ‡ä»¤)å¯«åœ¨è£¡é¢ï¼Œ æ­é…æ­£è¦è¡¨ç¤ºæ³•ã€ç®¡ç·šå‘½ä»¤èˆ‡è³‡æ–™æµé‡å°å‘ç­‰åŠŸèƒ½ï¼Œä»¥é”åˆ°æˆ‘å€‘æ‰€æƒ³è¦çš„è™•ç†ç›®çš„ï¼Œé€šéç·¨å¯«å’Œä½¿ç”¨ Shell è‰ç¨¿ï¼Œå¯ä»¥æé«˜å·¥ä½œæ•ˆç‡ä¸¦æ¸›å°‘é‡è¤‡æ“ä½œå‘½ä»¤
> - å¯åœ¨ Unixã€Linux æˆ–å…¶ä»–é¡ Unix ç³»çµ±çš„ shell ç’°å¢ƒä¸­é‹è¡Œã€‚æœ€å¸¸ç”¨çš„ shell è‰ç¨¿èªè¨€åŒ…æ‹¬ Bashï¼ˆBourne Again Shellï¼‰ã€Shï¼ˆBourne Shellï¼‰å’Œ Zshï¼ˆZ Shellï¼‰ã€‚

> reference:https://linux.vbird.org/linux_basic/centos7/0340bashshell-scripts.php#script

 
1. é€²å…¥resultè³‡æ–™å¤¾ï¼Œè¼¸å…¥`cd /work/username/result`
2. é€²å…¥shell scriptï¼Œè¼¸å…¥`vim fastqc.sh`

> [!IMPORTANT]
> #### å‘½ä»¤å°å­¸å ‚
> - åœ¨ Unix å’Œ Linux ç³»çµ±ä¸­ï¼Œå‘½ä»¤ `vim fastqc.sh` ç”¨æ–¼ä½¿ç”¨ Vim ç·¨è¼¯å™¨æ‰“é–‹æˆ–å‰µå»ºåç‚º `fastqc.sh` çš„æ–‡ä»¶ã€‚é€™æ˜¯ä¸€å€‹ç°¡å–®çš„å‘½ä»¤ï¼Œç”¨æ–¼é€²å…¥ Vim ç·¨è¼¯å™¨ä»¥ç·¨è¼¯æŒ‡å®šçš„ Shell Script æ–‡ä»¶ã€‚
> - å…·é«”å«ç¾©
>  * vimï¼šé€™æ˜¯å‘½ä»¤è¡Œæ–‡æœ¬ç·¨è¼¯å™¨ Vim çš„åç¨±ã€‚Vim æ˜¯ä¸€å€‹å¼·å¤§çš„æ–‡æœ¬ç·¨è¼¯å™¨ï¼Œå»£æ³›ç”¨æ–¼ç·¨è¼¯ç¨‹åºä»£ç¢¼å’Œè‰ç¨¿ã€‚
>  * `fastqc.sh`ï¼šé€™æ˜¯è¦ç·¨è¼¯çš„æ–‡ä»¶åã€‚åœ¨é€™ç¨®æƒ…æ³ä¸‹ï¼Œ`fastqc.sh` æ˜¯ä¸€å€‹ Shell Script æ–‡ä»¶ï¼Œæ“´å±•å .sh é€šå¸¸è¡¨ç¤ºé€™æ˜¯ä¸€å€‹ Shell è„šæœ¬æ–‡ä»¶ã€‚
> - åŸ·è¡Œè©²å‘½ä»¤å¾Œçš„è¡Œç‚ºï¼š
>  * å¦‚æœ `fastqc.sh` æ–‡ä»¶å·²ç¶“å­˜åœ¨ï¼Œ`vim fastqc.sh` æœƒæ‰“é–‹é€™å€‹æ–‡ä»¶ï¼Œå…è¨±ä½ æŸ¥çœ‹å’Œç·¨è¼¯å…¶å…§å®¹ã€‚
>  * å¦‚æœ `fastqc.sh` æ–‡ä»¶ä¸å­˜åœ¨ï¼Œ`vim fastqc.sh` æœƒå‰µå»ºä¸€å€‹æ–°çš„ç©ºæ–‡ä»¶ï¼Œä¸¦é€²å…¥ Vim ç·¨è¼¯å™¨ä»¥ä¾¿ä½ å¯ä»¥é–‹å§‹ç·¨å¯«è‰ç¨¿ã€‚

  
3. è«‹æŒ‰éµç›¤ <kbd>i</kbd> é€²å…¥ç·¨è¼¯æ¨¡å¼ (åº•ä¸‹æœƒå‡ºç¾"â€“- INSERT â€“-")ï¼Œä¸¦æŠŠä¸Šä¸€æ­¥é©Ÿä¸‹è¼‰çš„shell scipt è²¼åˆ°`fastqc.sh`


## Step 2 ä¿®æ”¹åˆ†æåŸ·è¡Œæª”

1. è«‹æ›´æ”¹ä»¥ä¸‹ç¨‹å¼ç¢¼ï¼š
> ä»¥ä¸‹ç¤ºç¯„æœƒä»¥fastqè³‡æ–™å¤¾ä¸­çš„SEAåšç‚ºç¤ºç¯„ (æ ¼å¼è«‹ä¾ç…§è£¡é¢çµ¦ä½ çš„ç¯„ä¾‹ï¼Œå‰¯æª”åä¸ç”¨å¯«é€²å»)
> 

(1) Slurmæ’ç¨‹è¨­å®š

> [!IMPORTANT]
> #### slurmæ˜¯ç”šéº¼?
> SLURMï¼ˆSimple Linux Utility for Resource Managementï¼‰æ˜¯ä¸€å€‹ç”¨æ–¼å¤§è¦æ¨¡è¨ˆç®—é›†ç¾¤çš„é–‹æºè³‡æºç®¡ç†å™¨å’Œå·¥ä½œè² è¼‰ç®¡ç†å™¨ã€‚å®ƒä¸»è¦ç”¨æ–¼é«˜æ€§èƒ½è¨ˆç®—ï¼ˆHPCï¼‰ç’°å¢ƒï¼Œå¹«åŠ©ç®¡ç†å’Œèª¿åº¦è¨ˆç®—è³‡æºï¼Œå¦‚ CPUã€å…§å­˜å’Œè¨ˆç®—ç¯€é»ã€‚SLURM åœ¨å¤§å‹è¶…ç®—ä¸­å¿ƒã€ç ”ç©¶æ©Ÿæ§‹å’Œä¼æ¥­ä¸­å»£æ³›ä½¿ç”¨ã€‚


  ğŸ‘‰ æ¥ä¸‹ä¾†ä¾ç…§æŒ‡ç¤ºä¿®æ”¹é€™å€‹å€å¡Š (è«‹è¦‹ä¸‹é¢èªªæ˜)ï¼š

  ```
  #!/usr/bin/sh
  #SBATCH -A ACD114093                      # Account name/project number
  #SBATCH -J fastqc                         # Job name
  #SBATCH -p ngscourse                      # Partition Name ç­‰åŒPBSè£¡é¢çš„ -q Queue name
  #SBATCH -c 2                              # ä½¿ç”¨çš„coreæ•¸ è«‹åƒè€ƒQueueè³‡æºè¨­å®š
  #SBATCH --mem=13g                         # ä½¿ç”¨çš„è¨˜æ†¶é«”é‡ è«‹åƒè€ƒQueueè³‡æºè¨­å®š
  #SBATCH -o out.log                        # Path to the standard output file
  #SBATCH -e err.log                        # Path to the standard error ouput file
  #SBATCH --mail-user=yourmail@gmail.com    # email
  #SBATCH --mail-type=END                   # æŒ‡å®šé€å‡ºemailæ™‚æ©Ÿ å¯ç‚ºNONE, BEGIN, END, FAIL, REQUEUE, ALL
  # åœ‹ç¶²ä½¿ç”¨
  ```

(2) ä¿®æ”¹æª”æ¡ˆè·¯å¾‘  
  ```
  # Please enter the R1 & R2 file name and your username
  sampleR1=/work/username/result/fastqc/SRR13076392_S14_L002_R1_001.fastq.gz    # Readsæ‰€åœ¨çš„æª”æ¡ˆè·¯å¾‘
  sampleR2=/work/username/result/fastqc/SRR13076392_S14_L002_R2_001.fastq.gz    # Readsæ‰€åœ¨çš„æª”æ¡ˆè·¯å¾‘
  ```

(3)å»ºç«‹è³‡æ–™å¤¾(å‘½åç‚º`fastqc_S14`)ä¾†å­˜æ”¾FastQCçµæœ
  ```
  mkdir fastqc_S14
  ```
    
(4) ä¿®æ”¹åŸ·è¡ŒFastQCçš„å‘½ä»¤
  ```
  fastqc ${sampleR1} ${sampleR2} -o fastqc_S14
  ```
(5)æŒ‰ <kbd>Esc</kbd> é›¢é–‹ç·¨è¼¯æ¨¡å¼  
(6) è¼¸å…¥ `:wq` ä¸¦æŒ‰ä¸‹ <kbd>Enter</kbd> å¯å„²å­˜çµæœ  
**â—è‹¥å‡ºç¾ "E45: 'readonly' option is set (add ! to override)" çš„è©±ï¼Œè«‹è¼¸å…¥`:wq!`ä¾†å„²å­˜ï¼‰â—**

> [!IMPORTANT]
> #### å‘½ä»¤å°å­¸å ‚
> - ```:wq```: ä¿å­˜ä¸¦é€€å‡º    
> æ˜¯åœ¨ Vim ç·¨è¼¯å™¨ä¸­ç”¨ä¾†ä¿å­˜æ–‡ä»¶ä¸¦é€€å‡ºç·¨è¼¯æ¨¡å¼çš„å‘½ä»¤ï¼Œå…ˆæŒ‰ ```Esc``` é€²å…¥å‘½ä»¤æ¨¡å¼ï¼Œç„¶å¾Œè¼¸å…¥ ```:wq``` ä¸¦æŒ‰ä¸‹ Enterã€‚é€™å°‡æœƒä¿å­˜ç•¶å‰æ–‡ä»¶çš„ä¿®æ”¹ä¸¦é€€å‡º Vimã€‚ ```w``` ä»£è¡¨ writeï¼ˆä¿å­˜ï¼‰ï¼Œ ```q``` ä»£è¡¨ quitï¼ˆé€€å‡ºï¼‰
> - ```:wq!```: å¼·åˆ¶ä¿å­˜ä¸¦é€€å‡º    
> å¦‚æœæ–‡ä»¶æ˜¯åªè®€çš„æˆ–è€…æœ‰å…¶ä»–é™åˆ¶ï¼Œå¯ä»¥ä½¿ç”¨ ```:wq!``` ä¾†å¼·åˆ¶ä¿å­˜ä¸¦é€€å‡ºã€‚ ```!``` æ˜¯å¼·åˆ¶åŸ·è¡Œçš„æ„æ€
> - å¦‚æœä½ åªæƒ³ä¿å­˜æ–‡ä»¶ä½†ä¸é€€å‡ºï¼Œå¯ä»¥ä½¿ç”¨ ```:w```ï¼Œå¦‚æœåªæƒ³é€€å‡ºä½†ä¸ä¿å­˜ï¼Œå¯ä»¥ä½¿ç”¨ ```:q``` æˆ– ```:q!``` (å¼·åˆ¶é€€å‡º)ã€‚

2. åŸ·è¡Œscript

(1)è¼¸å…¥ä»¥ä¸‹æŒ‡ä»¤ï¼Œä¾†ä»¥sbatch jobçš„æ–¹å¼é€å‡ºç·¨è¼¯å®Œæˆçš„è‰ç¨¿
```
sbatch fastqc.sh
```


> [!IMPORTANT]
> #### å‘½ä»¤å°å­¸å ‚
> ```sbatch``` æ˜¯ SLURM çš„ä¸€å€‹å‘½ä»¤è¡Œå·¥å…·ï¼Œç”¨æ–¼æäº¤ä½œæ¥­è‰ç¨¿åˆ° SLURM ä½œæ¥­èª¿åº¦ç³»çµ±ã€‚é€™äº›è‰ç¨¿é€šå¸¸åŒ…å« SLURM æŒ‡ä»¤å’Œè¦åŸ·è¡Œçš„å‘½ä»¤ã€‚

(2)è‹¥é€å‡ºæˆåŠŸå°‡æœƒå‡ºç¾ä»¥ä¸‹æ–‡å­—(çµæœåœ¨resultè³‡æ–™å¤¾å·²ç¶“æŒ‡å®šå¥½è·¯å¾‘)
```
Submitted batch job ï¼¿ï¼¿ï¼¿
```

(3)å¯ä½¿ç”¨ä»¥ä¸‹æŒ‡ä»¤æŸ¥çœ‹å·¥ä½œåŸ·è¡Œæƒ…æ³
```
sacct
```
![image](https://hackmd.io/_uploads/Bkevfu630.png)


> [!IMPORTANT]
> #### å‘½ä»¤å°å­¸å ‚
> ``sacct`` æ­¤æŒ‡ä»¤ç”¨æ–¼åˆ—å‡ºå¸³è™Ÿçš„ç›¸é—œä»»å‹™æˆ–ä»»å‹™é›†ä¹‹ç‹€æ…‹ï¼Œä¾‹å¦‚é‹è¡Œä¸­ã€å·²çµ‚æ­¢æˆ–æ˜¯å·²å®Œæˆï¼Œæ˜¯æœ€åŸºæœ¬çš„æª¢è¦–ä»»å‹™æŒ‡ä»¤ã€‚å®ƒå¯ä»¥é¡¯ç¤ºä¾‹å¦‚IDã€ä½¿ç”¨è€…ã€ç‹€æ…‹ã€ä½¿ç”¨çš„è³‡æºç­‰è³‡è¨Šï¼Œé€™å€‹å‘½ä»¤å°æ–¼è¿½è¹¤å’Œåˆ†æä½œæ¥­çš„é‹è¡Œæƒ…æ³éå¸¸æœ‰ç”¨ã€‚

3. æŸ¥çœ‹çµæœ  
 `out.log`å’Œ`err.log`ç‚ºåŸ·è¡Œé€™å€‹scriptçš„æ¨™æº–è¼¸å‡ºå’Œæ¨™æº–éŒ¯èª¤ï¼Œå¦‚æœåŸ·è¡Œæ™‚æœ‰å‡ºç¾éŒ¯èª¤ï¼Œå¯ä»¥æŸ¥çœ‹`err.log`(æª”æ¡ˆæœƒåœ¨`/work/username/result/`åº•ä¸‹)
* é€™ä»½åŸ·è¡Œæª”æœƒç”¢ç”Ÿhtmlæª”ï¼Œä¸‹è¼‰å¾Œå³å¯é–‹å•ŸæŸ¥çœ‹FastQC Report
* ä¸‹è¼‰è©³æƒ…å¯è¦‹é€£çµ[ä¸‹è¼‰](https://github.com/Jacob-s-Lab/2025-Biomarkers/blob/main/tutorial%20week1.md)
  

